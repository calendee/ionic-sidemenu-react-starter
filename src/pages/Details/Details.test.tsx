import React from 'react';
import { render } from '../../../test/test-utils';

import DetailsPage from './Details';

const defaultProps = {
  match: {
    params: {
      id: 20,
    },
  },
};

describe('Details Page', () => {
  test('renders and matches snapshot', () => {
    const { container, getByText } = render(<DetailsPage {...defaultProps} />);
    const {
      match: {
        params: { id },
      },
    } = defaultProps;
    const regEx = new RegExp(`you clicked on list item ${id}.`, 'i');
    getByText(regEx);
    expect(container).toMatchSnapshot();
  });
});
