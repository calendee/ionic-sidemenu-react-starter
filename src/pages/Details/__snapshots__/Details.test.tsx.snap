// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details Page renders and matches snapshot 1`] = `
<div>
  <div
    class="ion-page"
    data-testid="details-page"
  >
    <ion-header>
      <ion-toolbar>
        <ion-buttons
          slot="start"
        >
          <ion-back-button
            default-href="/home/list"
          />
        </ion-buttons>
        <ion-title>
          Details 
          20
        </ion-title>
      </ion-toolbar>
    </ion-header>
    <ion-content>
      <p>
        The devil is in the details. You clicked on list item 
        20
        .
      </p>
    </ion-content>
  </div>
</div>
`;
